pylint/message/message_id_store.py

```diff
@@ -101,18 +102,25 @@ def _raise_duplicate_msgid(symbol: str, msgid: str, other_msgid: str) -> NoRetur
         )
         raise InvalidMessageError(error_message)
 
+    @functools.lru_cache()
     def get_active_msgids(self, msgid_or_symbol: str) -> List[str]:
-        """Return msgids but the input can be a symbol."""
-        # Only msgid can have a digit as second letter
-        is_msgid: bool = msgid_or_symbol[1:].isdigit()
-        msgid = None
-        if is_msgid:
+        """Return msgids but the input can be a symbol.
+
+        The cache has no limit as it will never go very high. It's a set number of
+        msgid/symbol pair from our own code and from user's checkers. How many msgid can
+        the user generate ? I guess it will never be more than 1000, and if each
+        msgid/symbol pair has around 50 characters, it's still ~= 5 kb to cache.
+        """
+        msgid: Optional[str]
+        symbol: Optional[str]
```

Comment:
Is this necessary? Or does `mypy` infer this itself?

Reply:
The issue here is that msgid is pure string line 118, but Optional[str] line 121

Reply:
We would have the same problem but with ``symbol`` if we change the order of condition.

Reply:
👍

Reply:
```suggestion
```